nms/packages/magmalte/fbc_js_core/ui/host/OrganizationEdit.js

```diff
@@ -181,205 +145,152 @@ export default function OrganizationEdit(props: Props) {
+          onClose={() => {
+            setAddUser(false);
+            setDialog(false);
+            setEditUser(false);
+          }}
+          onCreateOrg={org => {
+            onSave(org);
+            setRefresh(Date.now());
+            setDialog(false);
+          }}
+          onCreateUser={newUser => {
+            if (!editUser) {
+              axios
+                .post(
+                  `/host/organization/async/${
+                    match.params.name || ''
+                  }/add_user`,
+                  newUser,
+                )
+                .then(() => {
+                  enqueueSnackbar('User added successfully', {
+                    variant: 'success',
+                  });
+                  setDialog(false);
+                })
+                .catch(error => {
+                  setCreateError(error);
+                  enqueueSnackbar(error?.response?.data?.error || error, {
+                    variant: 'error',
+                  });
+                });
+            } else {
+              if (user?.id) {
+                axios
+                  .put(`/user/async/${user.id}`, newUser)
+                  .then(() => {
+                    enqueueSnackbar('User updated successfully', {
+                      variant: 'success',
+                    });
+                    setDialog(false);
+                  })
+                  .catch(error => {
+                    setCreateError(error);
+                    enqueueSnackbar(error?.response?.data?.error || newUser, {
+                      variant: 'error',
+                    });
+                  });
+              }
+            }
+          }}
+        />
+        <Grid container spacing={4}>
+          <Grid container justify="space-between">
+            <Grid item>
+              <Grid container alignItems="center">
+                <Grid>
+                  <IconButton
+                    onClick={() => history.goBack()}
+                    className={classes.arrowBack}
+                    color="primary">
+                    <ArrowBackIcon />
+                  </IconButton>
+                </Grid>
+                <Grid>
+                  <Text className={classes.header} variant="h4">
+                    {organization?.name}
+                  </Text>
+                </Grid>
+              </Grid>
+            </Grid>
+            <Grid item>
+              <Button
+                skin="comet"
+                onClick={() => {
+                  props
+                    .confirm(
+                      'Are you sure you want to delete this organization?',
+                    )
+                    .then(async confirm => {
+                      if (!confirm) return;
+                      await axios.delete(
+                        `/host/organization/async/${organization?.id}`,
+                      );
+                      history.push('/host/organizations');
+                    });
```

Comment:
Bit of a nit, but could you move this logic out of the returned JSX? It makes it a bit easier to read through.